{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"CNN_mnist_ing.ipynb","provenance":[],"collapsed_sections":[]},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"code","metadata":{"id":"stKU9gXA-3U8","colab_type":"code","outputId":"2579d59e-c0fc-4db1-acff-a7b5ebb8da23","executionInfo":{"status":"ok","timestamp":1569995054757,"user_tz":-540,"elapsed":757,"user":{"displayName":"eunsu Kim","photoUrl":"","userId":"02390913583849598770"}},"colab":{"base_uri":"https://localhost:8080/","height":89}},"source":["import os,warnings\n","warnings.filterwarnings(action='ignore')\n","\n","from tensorflow.examples.tutorials.mnist import input_data\n","mnist=input_data.read_data_sets('./mnist/data/',one_hot=True)\n","\n","import tensorflow as tf"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Extracting ./mnist/data/train-images-idx3-ubyte.gz\n","Extracting ./mnist/data/train-labels-idx1-ubyte.gz\n","Extracting ./mnist/data/t10k-images-idx3-ubyte.gz\n","Extracting ./mnist/data/t10k-labels-idx1-ubyte.gz\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"GCt27cJL_Nwt","colab_type":"code","colab":{}},"source":["X=tf.placeholder(tf.float32,[None,28,28,1])\n","Y=tf.placeholder(tf.float32,[None,10])\n","\n","is_training=tf.placeholder(tf.bool)"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"LkwjLuSC_y6b","colab_type":"code","outputId":"475a872a-e6a3-4a12-d196-ce405a3f952c","executionInfo":{"status":"ok","timestamp":1569995177557,"user_tz":-540,"elapsed":495,"user":{"displayName":"eunsu Kim","photoUrl":"","userId":"02390913583849598770"}},"colab":{"base_uri":"https://localhost:8080/","height":361}},"source":["L1 = tf.layers.conv2d(X,32,[3,3])\n","L1 = tf.layers.max_pooling2d(L1, [2,2], [2,2])\n","L1 = tf.layers.dropout(L1, 0.7, is_training)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["WARNING:tensorflow:From <ipython-input-8-04bd74df5026>:1: conv2d (from tensorflow.python.layers.convolutional) is deprecated and will be removed in a future version.\n","Instructions for updating:\n","Use `tf.keras.layers.Conv2D` instead.\n","WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/init_ops.py:1251: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n","Instructions for updating:\n","Call initializer instance with the dtype argument instead of passing it to the constructor\n","WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x7efc69f83dd8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x7efc69f83dd8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x7efc69f83dd8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x7efc69f83dd8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING:tensorflow:From <ipython-input-8-04bd74df5026>:2: max_pooling2d (from tensorflow.python.layers.pooling) is deprecated and will be removed in a future version.\n","Instructions for updating:\n","Use keras.layers.MaxPooling2D instead.\n","WARNING:tensorflow:Entity <bound method Pooling2D.call of <tensorflow.python.layers.pooling.MaxPooling2D object at 0x7efc691ee6a0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Pooling2D.call of <tensorflow.python.layers.pooling.MaxPooling2D object at 0x7efc691ee6a0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING: Entity <bound method Pooling2D.call of <tensorflow.python.layers.pooling.MaxPooling2D object at 0x7efc691ee6a0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Pooling2D.call of <tensorflow.python.layers.pooling.MaxPooling2D object at 0x7efc691ee6a0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING:tensorflow:From <ipython-input-8-04bd74df5026>:3: dropout (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.\n","Instructions for updating:\n","Use keras.layers.dropout instead.\n","WARNING:tensorflow:Entity <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc69f83dd8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc69f83dd8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING: Entity <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc69f83dd8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc69f83dd8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"tdIM_2SrATvu","colab_type":"code","outputId":"54889d23-3780-4814-ef4a-0a6696507a29","executionInfo":{"status":"ok","timestamp":1569995204007,"user_tz":-540,"elapsed":482,"user":{"displayName":"eunsu Kim","photoUrl":"","userId":"02390913583849598770"}},"colab":{"base_uri":"https://localhost:8080/","height":145}},"source":["L2 = tf.layers.conv2d(L1, 64, [3,3])\n","L2 = tf.layers.max_pooling2d(L2, [2,2], [2,2])\n","L2 = tf.layers.dropout(L2, 0.7, is_training)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x7efc691eedd8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x7efc691eedd8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x7efc691eedd8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x7efc691eedd8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING:tensorflow:Entity <bound method Pooling2D.call of <tensorflow.python.layers.pooling.MaxPooling2D object at 0x7efc692a7e10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Pooling2D.call of <tensorflow.python.layers.pooling.MaxPooling2D object at 0x7efc692a7e10>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING: Entity <bound method Pooling2D.call of <tensorflow.python.layers.pooling.MaxPooling2D object at 0x7efc692a7e10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Pooling2D.call of <tensorflow.python.layers.pooling.MaxPooling2D object at 0x7efc692a7e10>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING:tensorflow:Entity <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc692a7cf8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc692a7cf8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING: Entity <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc692a7cf8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc692a7cf8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"MhJN421aAaNX","colab_type":"code","outputId":"7dfca7ec-8b24-4568-bffb-08947de36b59","executionInfo":{"status":"ok","timestamp":1569995304018,"user_tz":-540,"elapsed":483,"user":{"displayName":"eunsu Kim","photoUrl":"","userId":"02390913583849598770"}},"colab":{"base_uri":"https://localhost:8080/","height":253}},"source":["L3 = tf.contrib.layers.flatten(L2)\n","L3 = tf.layers.dense(L3, 256, activation=tf.nn.relu)\n","L3 = tf.layers.dropout(L3, 0.5, is_training)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/contrib/layers/python/layers/layers.py:1634: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.\n","Instructions for updating:\n","Use keras.layers.flatten instead.\n","WARNING:tensorflow:Entity <bound method Flatten.call of <tensorflow.python.layers.core.Flatten object at 0x7efc691eedd8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Flatten.call of <tensorflow.python.layers.core.Flatten object at 0x7efc691eedd8>>: AttributeError: module 'gast' has no attribute 'Num'\n","WARNING: Entity <bound method Flatten.call of <tensorflow.python.layers.core.Flatten object at 0x7efc691eedd8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Flatten.call of <tensorflow.python.layers.core.Flatten object at 0x7efc691eedd8>>: AttributeError: module 'gast' has no attribute 'Num'\n","WARNING:tensorflow:From <ipython-input-10-ccdf79516e01>:2: dense (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.\n","Instructions for updating:\n","Use keras.layers.dense instead.\n","WARNING:tensorflow:Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x7efc69f83400>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x7efc69f83400>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING: Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x7efc69f83400>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x7efc69f83400>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING:tensorflow:Entity <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc69f83ba8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc69f83ba8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING: Entity <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc69f83ba8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dropout.call of <tensorflow.python.layers.core.Dropout object at 0x7efc69f83ba8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"Y25YibQ-AyoG","colab_type":"code","outputId":"02dcea69-88fc-4450-bc6d-f5ef92cf18ba","executionInfo":{"status":"ok","timestamp":1569995371927,"user_tz":-540,"elapsed":488,"user":{"displayName":"eunsu Kim","photoUrl":"","userId":"02390913583849598770"}},"colab":{"base_uri":"https://localhost:8080/","height":73}},"source":["model = tf.layers.dense(L3, 10, activation=None)\n","\n"],"execution_count":0,"outputs":[{"output_type":"stream","text":["WARNING:tensorflow:Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x7efc691ee390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x7efc691ee390>>: AssertionError: Bad argument number for Name: 3, expecting 4\n","WARNING: Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x7efc691ee390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x7efc691ee390>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"ASd1TwOdBDNO","colab_type":"code","colab":{}},"source":["cost = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits_v2(logits=model, labels=Y))\n","optimizer = tf.train.AdamOptimizer(0.001).minimize(cost)"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"18XkVrvRBFIW","colab_type":"code","outputId":"20dfe48e-c3d8-4005-b99b-948b12b13053","executionInfo":{"status":"ok","timestamp":1569995827794,"user_tz":-540,"elapsed":356452,"user":{"displayName":"eunsu Kim","photoUrl":"","userId":"02390913583849598770"}},"colab":{"base_uri":"https://localhost:8080/","height":251}},"source":["%%time\n","\n","init = tf.global_variables_initializer()\n","sess = tf.Session()\n","sess.run(init)\n","\n","batch_size = 100\n","total_batch = int(mnist.train.num_examples / batch_size)\n","\n","for epoch in range(10):\n","    total_cost = 0\n","    for i in range(total_batch):\n","        batch_xs, batch_ys = mnist.train.next_batch(batch_size)\n","\n","        batch_xs = batch_xs.reshape(-1, 28, 28, 1)\n","        _, cost_val = sess.run([optimizer, cost],\n","                       feed_dict={X: batch_xs,\n","                                  Y: batch_ys,\n","                                  is_training: True})\n","        total_cost += cost_val\n","    print('Epoch:', '%04d' % (epoch + 1),\n","          'Avg. cost =', '{:.3f}'.format(total_cost / total_batch))\n","print('최적화 완료!')"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Epoch: 0001 Avg. cost = 0.176\n","Epoch: 0002 Avg. cost = 0.049\n","Epoch: 0003 Avg. cost = 0.031\n","Epoch: 0004 Avg. cost = 0.022\n","Epoch: 0005 Avg. cost = 0.017\n","Epoch: 0006 Avg. cost = 0.013\n","Epoch: 0007 Avg. cost = 0.012\n","Epoch: 0008 Avg. cost = 0.011\n","Epoch: 0009 Avg. cost = 0.006\n","Epoch: 0010 Avg. cost = 0.006\n","최적화 완료!\n","CPU times: user 11min 23s, sys: 9.26 s, total: 11min 32s\n","Wall time: 5min 56s\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"Xb8h6ITfBH92","colab_type":"code","colab":{}},"source":[""],"execution_count":0,"outputs":[]}]}